src/Fixer/PhpUnit/PhpUnitDataProviderMethodOrderFixer.php

```diff
@@ -0,0 +1,288 @@
+use PhpCsFixer\Fixer\ConfigurableFixerTrait;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Analyzer\DataProviderAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @implements ConfigurableFixerInterface<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration>
+ *
+ * @phpstan-type _AutogeneratedInputConfiguration array{
+ *  placement?: 'after'|'before'
+ * }
+ * @phpstan-type _AutogeneratedComputedConfiguration array{
+ *  placement: 'after'|'before'
+ * }
+ *
+ * @phpstan-import-type _ClassElement from OrderedClassElementsFixer
+ */
+final class PhpUnitDataProviderMethodOrderFixer extends AbstractPhpUnitFixer implements ConfigurableFixerInterface
+{
+    /** @use ConfigurableFixerTrait<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration> */
+    use ConfigurableFixerTrait;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Data provider method must be placed after/before the first test where used.',
+            [
+                new CodeSample(
+                    '<?php
+class FooTest extends TestCase {
+    public function dataProvider() {}
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testSomething($expected, $actual) {}
+}
+',
+                ),
+                new CodeSample(
+                    '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testSomething($expected, $actual) {}
+    public function dataProvider() {}
+}
+',
+                    [
+                        'placement' => 'before',
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before ClassAttributesSeparationFixer, NoBlankLinesAfterClassOpeningFixer.
+     * Must run after OrderedClassElementsFixer.
+     */
+    public function getPriority(): int
+    {
+        return 64;
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('placement', 'Prefix that replaces "test".'))
+                ->setAllowedValues(['after', 'before'])
+                ->setDefault('after')
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyPhpUnitClassFix(Tokens $tokens, int $startIndex, int $endIndex): void
+    {
+        $elements = $this->getElements($tokens, $startIndex);
+
+        if (0 === \count($elements)) {
+            return;
+        }
+
+        $endIndex = $elements[\count($elements) - 1]['end'];
```

Comment:
```suggestion
        $endIndex = $elements[array_key_last($elements)]['end'];
```

Reply:
👍 adjusted here and in `OrderedClassElementsFixer`